### PR TITLE
Fix shellcheck failures in test/cmd/g.*sh

### PR DESCRIPTION
**Pull request by xychu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Part of #72956
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This is a follow-up pr of #74639 and it is part of fixes under test/cmd/*.sh
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/cmd/generic-resources.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ run_multi_resources_tests() {
   YAML=".yaml"
   JSON=".json"
   for file in $FILES; do
-    if [ -f $file$YAML ]
+    if [ -f "$file"$YAML ]
```
Comment (BenTheElder) [inline test/cmd/generic-resources.sh]: Quote the whole string and use brace expansion?

File: test/cmd/generic-resources.sh [hunk 2]
```diff
@@ -298,7 +299,7 @@ run_recursive_resources_tests() {
   kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" 'busybox0:busybox1:'
   # Command
   output_message=$(! kubectl label -f hack/testdata/recursive/pod mylabel='myvalue' --recursive 2>&1 "${kube_flags[@]}")
-  echo $output_message
+  echo "$output_message"
```
Comment (BenTheElder) [inline test/cmd/generic-resources.sh]: Nit: we should be using ${} consistently while we're at it.

File: test/cmd/get.sh [hunk 3]
```diff
@@ -138,9 +138,9 @@ run_kubectl_get_tests() {
 
   ### Test kubectl get chunk size does not result in a --watch error when resource list is served in multiple chunks
   # Pre-condition: ConfigMap one two tree does not exist
-  kube::test::get_object_assert 'configmaps' '{{range.items}}{{ if eq $id_field \"one\" }}found{{end}}{{end}}:' ':'
-  kube::test::get_object_assert 'configmaps' '{{range.items}}{{ if eq $id_field \"two\" }}found{{end}}{{end}}:' ':'
-  kube::test::get_object_assert 'configmaps' '{{range.items}}{{ if eq $id_field \"three\" }}found{{end}}{{end}}:' ':'
+  kube::test::get_object_assert 'configmaps' "{{range.items}}{{ if eq $id_field \\\"one\\\" }}found{{end}}{{end}}:" ':'
+  kube::test::get_object_assert 'configmaps' "{{range.items}}{{ if eq $id_field \\\"two\\\" }}found{{end}}{{end}}:" ':'
+  kube::test::get_object_assert 'configmaps' "{{range.items}}{{ if eq $id_field \\\"three\\\" }}found{{end}}{{end}}:" ':'
```
Comment (BenTheElder) [inline test/cmd/get.sh]: I _think_ this is fine, but it does change when the variable is resolved.

File: test/cmd/generic-resources.sh [hunk 4]
```diff
@@ -308,7 +309,7 @@ run_recursive_resources_tests() {
   kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" 'busybox0:busybox1:'
   # Command
   output_message=$(! kubectl patch -f hack/testdata/recursive/pod -p='{"spec":{"containers":[{"name":"busybox","image":"prom/busybox"}]}}' --recursive 2>&1 "${kube_flags[@]}")
-  echo $output_message
+  echo "$output_message"
```
Comment (BenTheElder) [inline test/cmd/generic-resources.sh]: Same nit

Comment (BenTheElder) [reply test/cmd/generic-resources.sh]: There are several other places where we are now quoting strings that we should do this.

Comment (xychu) [reply test/cmd/generic-resources.sh]: @BenTheElder all Updated.

Comment (BenTheElder) [reply test/cmd/generic-resources.sh]: thank you :pray:

File: test/cmd/generic-resources.sh [hunk 5]
```diff
@@ -33,12 +33,12 @@ run_multi_resources_tests() {
   YAML=".yaml"
   JSON=".json"
   for file in $FILES; do
-    if [ -f $file$YAML ]
+    if [ -f "${file}${YAML}" ]
     then
-      file=$file$YAML
+      file=${file}${YAML}
```
Comment (BenTheElder) [inline test/cmd/generic-resources.sh]: shouldn't these be quoted?

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
possibly should quote a bit more, but this is an improvement, thanks!

**Pull request merged**